beamer/util.py

```diff
@@ -165,9 +172,18 @@ def __init__(self, tokens: List[List[List[str]]]) -> None:
                      Please check {l2_chain_ids}"""
                     assert intersection == l2_chain_ids, msg
 
-            for token in equiv_class:
+            for token in token_mapping:
+                chain_id = ChainId(int(token[0]))
+                token_address = to_checksum_address(token[1])
+                if len(token) == 2:
+                    allowance = None
+                elif int(token[2]) == -1:
+                    allowance = 2**256 - 1
+                else:
+                    allowance = int(token[2])
```

Comment:
I always wanted to try out the new structural pattern matching feature of recent Python and this looks like a perfect place for it! :smile: 

Consider
```python
match token:
   case [_, _]:
    allowance = None
   case [_, _, -1]:
    allowance = 2**256 - 1
   case [_, _, allowance]:
    pass # not sure whether int conversion is really needed here -- won't toml convert it to an int?
   case _:
    raise ValueError('unexpected token data: %r', token)
```

Reply:
>Also, not sure whether we check it already somewhere, but it would be good if we check that the allowance is >= 0 (if it is not -1).

I think it is a general problem that we do not really check the configured values. So we should address this as a whole IMO. ( I think there is already an issue to that)